### PR TITLE
AJ-112 add authentification to classification mailet

### PR DESCRIPTION
**Pull request by strokyl**

### CODE REVIEW

File: src/main/java/com/linagora/james/mailets/GuessClassificationMailet.java [hunk 1]
```diff
@@ -141,6 +150,30 @@ public void init() throws MessagingException {
...
+        if (Strings.isNullOrEmpty(serviceUsername)) {
+            throw new MailetException("'serviceUsername' is mandatory");
+        }
+
+        if (Strings.isNullOrEmpty(servicePassword)) {
+            throw new MailetException("'servicePassword' is mandatory");
+        }
+
+        try {
+            URIBuilder uriBuilder = new URIBuilder(serviceUrl);
```
Comment (aduprat) [inline src/main/java/com/linagora/james/mailets/GuessClassificationMailet.java]: Extract this part to a method.

**Comment by aduprat** [comment]
It could be great to have a new test class for this mailet based on an authenticated service in docker.
Then you will be able to check if authentication works.

File: src/main/java/com/linagora/james/mailets/GuessClassificationMailet.java [hunk 2]
```diff
@@ -141,6 +150,30 @@ public void init() throws MessagingException {
...
+            HttpHost host = new HttpHost(uriBuilder.getHost(), uriBuilder.getPort(), uriBuilder.getScheme());
+            executor = Executor.newInstance()
+                    .authPreemptive(host)
+                    .auth(host, new UsernamePasswordCredentials(serviceUsername, servicePassword));
+
+            headerName = getInitParameter(HEADER_NAME, HEADER_NAME_DEFAULT_VALUE);
```
Comment (strokyl) [inline src/main/java/com/linagora/james/mailets/GuessClassificationMailet.java]: Fixed sorry

File: src/main/java/com/linagora/james/mailets/GuessClassificationMailet.java (hunk 1 above)
Comment (strokyl) [inline src/main/java/com/linagora/james/mailets/GuessClassificationMailet.java]: Done

File: src/main/java/com/linagora/james/mailets/GuessClassificationMailet.java [hunk 3]
```diff
@@ -121,15 +132,13 @@ public GuessClassificationMailet() {
...
         LOGGER.debug("init GuessClassificationMailet");
-        executorService = Executors.newFixedThreadPool(
-            MailetUtil.getInitParameterAsStrictlyPositiveInteger(
-                getInitParameter(THREAD_COUNT),
-                THREAD_COUNT_DEFAULT_VALUE));
-
         timeoutInMs = parseTimeout();
 
         serviceUrl = getInitParameter(SERVICE_URL);
+        serviceUsername = getInitParameter(SERVICE_USERNAME);
```
Comment (strokyl) [inline src/main/java/com/linagora/james/mailets/GuessClassificationMailet.java]: Done

**Comment by strokyl** [comment]
> It could be great to have a new test class for this mailet based on an authenticated service in docker.
Then you will be able to check if authentication works.

IMO this is overkill, this is just basic http authentification and we aren't handling this part ourself, we rely on apache http client. This will be testing apache http client code

**Comment by aduprat** [comment]
I agree with you, it's not easy; but we can have two simple tests:
- bad credentials -> 401
- good credentials -> 404 (no endpoint)

**Pull request merged**